cmd/clusterctl/client/cluster/template.go

```diff
@@ -305,7 +305,8 @@ func getGitHubClient(ctx context.Context, configVariablesClient config.Variables
 
 // handleGithubErr wraps error messages.
 func handleGithubErr(err error, message string, args ...interface{}) error {
-	if _, ok := err.(*github.RateLimitError); ok {
+	var rateLimitErr *github.RateLimitError
+	if errors.As(err, &rateLimitErr) {
```

Comment:
```suggestion
	if errors.Is(err, &github.RateLimitError{}) {
```

Can we please do this instead for all the cases where we don't use the error value?

Reply:
fixed.